debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnectorConfig.java

```diff
@@ -221,6 +221,15 @@ public static SnapshotIsolationMode parse(String value, String defaultValue) {
     public static final Field PORT = RelationalDatabaseConnectorConfig.PORT
             .withDefault(DEFAULT_PORT);
 
+    public static final Field NOPORT = Field.create(DATABASE_CONFIG_PREFIX + SqlServerConnection.NOPORT_NAME)
```

Comment:
@kyleyj I am not happy with this option. I'd prefer to modify `PORT` locally to be optional instead of required. In that case it would be just enough to check if it is present or not and there will be no need for new config option.

Reply:
I would have preferred that option too, but that would have introduced a breaking change for anyone that was expecting the default port (1433) to be used when not provided.  In fact, the implementation of defaulting to default port when not provided doesn't following the specification for JDBC.  So, I would prefer to modify `PORT` to be optional but others would need to weigh in on how impactful that change would be.

Reply:
@kyleyj That's very good point. But how about using a magical umber like `0`? If set to `0` then use named instance approach?

Reply:
Using a magic number (0, -1, etc) would work, but still requires knowledge of the intended behavior like the `noport` property.  How have breaking changes been communicated in the past?  I've noticed that with v19/v20, there was a change with some of the other properties (i.e. database.server.name, database.names, etc.).  Would we just change/communicate the change of defaulting to 1433 when `PORT` is not provided?

Reply:
@kyleyj We are swiching a major version so brekaing changes are allowed. Still I believe that your original argument has a merit. Let's stay with `1433` as default for two reasons

- it is more frequent scenario
- it is consistent with behaviour of other connectors

From user PoV knowledge there is no difference between using new option or magic number. In both cases the docs need to be read. Yet I'd personally prefer to use magical number as the new option is introduced to disable another option which is a bit strange.

Thinking about that there is even one more solution - a generic one that we provide for Oracle - https://debezium.io/documentation/reference/1.9/connectors/oracle.html#oracle-property-database-url

Reply:
@kyleyj And the thing I completely forget about originally - please see https://issues.redhat.com/browse/DBZ-4001

We can provide OOTB alternative connection factory that would just not use the port setting.

Reply:
That looks like a great option and is probably the right approach.  I'll be testing this out tomorrow.  However, I'm not able to see any documentation that would have led me to this option in the first place.  Other than looking at the example from the ticket, where is the documentation to support this feature?

Reply:
Good point, we probably forgot to add it to the docs. The in-code description is loated in https://github.com/debezium/debezium/blob/main/debezium-core/src/main/java/io/debezium/jdbc/JdbcConfiguration.java#L64